source/common/network/listener_impl.cc

```diff
@@ -14,36 +14,34 @@
 
 namespace Network {
 
-void ListenerImpl::listenCallback(evconnlistener*, evutil_socket_t fd, sockaddr* addr, int,
+void ListenerImpl::listenCallback(evconnlistener*, evutil_socket_t fd, sockaddr* remote_addr, int,
                                   void* arg) {
   ListenerImpl* listener = static_cast<ListenerImpl*>(arg);
 
-  if (listener->use_original_dst_) {
-    sockaddr_storage orig_dst_addr;
-    memset(&orig_dst_addr, 0, sizeof(orig_dst_addr));
+  sockaddr_storage local_addr;
+  memset(&local_addr, 0, sizeof(local_addr));
 
-    bool success = Utility::getOriginalDst(fd, &orig_dst_addr);
+  bool success = Utility::getOriginalDst(fd, &local_addr);
```

Comment:
What does SO_ORIGINAL_DST return if it wasn't redirected and the success if false? Is is the true address? Or will it contain just an empty address?